### PR TITLE
Fix PostgreSQL set operation precedence

### PR DESCRIPTION
**Pull request by tisonkun**
Comes from https://github.com/antlr/grammars-v4/pull/2378#discussion_r1246216426.

PG treats set ops in different precedence - INTERSECT is tighter than UNION and EXCEPT.

### CODE REVIEW

**Comment by tisonkun** [comment]
test locally with `mvn test` under pg path and it passes all.

**Comment by tisonkun** [comment]
@teverett is Windows build unstable? I check the CI log and it seems all about package installing instead of tests failure.

File: sql/postgresql/PostgreSQLParser.g4 [hunk 1]
```diff
@@ -2902,33 +2902,25 @@ select_no_parens
...
 
 select_clause
-   : simple_select
-   | select_with_parens
+   : simple_select_intersect ((UNION | EXCEPT) all_or_distinct simple_select_intersect)*
    ;
 
-simple_select
+simple_select_intersect
+    : simple_select_pramary ((INTERSECT) all_or_distinct simple_select_pramary)*
```
Comment (KvanTTT) [inline sql/postgresql/PostgreSQLParser.g4:2909]: Useless parentheses:

```suggestion
    : simple_select_pramary (INTERSECT all_or_distinct simple_select_pramary)*
```

**Review by KvanTTT — CHANGES_REQUESTED** [review]
Please add a test that covers suggested changes.

**Comment by tisonkun** [comment]
> Please add a test that covers suggested changes.

The current maven tests seems only test if the query parsed, not the result it parsed out. Shall I build the whole test kit from scratch to verify the op precedence result, or there are some existing tools/cases to leverage?

**Review by KvanTTT — APPROVED** [review]

File: sql/postgresql/PostgreSQLParser.g4 [hunk 2]
```diff
@@ -2902,33 +2902,25 @@ select_no_parens
...
-   | select_with_parens
+   : simple_select_intersect ((UNION | EXCEPT) all_or_distinct simple_select_intersect)*
    ;
 
-simple_select
+simple_select_intersect
+    : simple_select_pramary (INTERSECT all_or_distinct simple_select_pramary)*
+    ;
+
+simple_select_pramary
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply sql/postgresql/PostgreSQLParser.g4:2912]: This should be a typo 🤣

**Pull request merged**